### PR TITLE
Pay 369 version control

### PR DESCRIPTION
**Pull request by StarfallProjects**

### CODE REVIEW

File: docs/source-control/patterns.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+This pattern is also useful when testing a new version of n8n: you can create a new n8n instance with the new version, connect it to the Git branch and test it, while your production instance remains on the older version until you're confident it's safe to upgrade.
+
+![Diagram](/_images/source-control/vc-multi-one.png)
+
+## One instance, multiple branches
+
+The instance owner can change which Git branch connects to the instance. The full setup in this case is likely to be a [Multiple instances, multiple branches](#multiple-instances-multiple-branches) pattern, but with one instance switching between branches.
+
+This is useful to review work. For example, different users could work on their own instance and push to their own branch. The reviewer could work in a review instance, and switch between branches to load work from different users.
```
Comment (krynble) [inline docs/source-control/patterns.md:35]: One minor problem I see with this setup is that when switching branches n8n won't "clean up" existing workflows.

This means that if we have branch dev with workflows A, B and C, and branch called "testing" with workflows B, C and D, when I switch from one branch to the other I'll always see all 4 workflows, as we never delete anything when pulling.

**Review by krynble — CHANGES_REQUESTED** [review]

File: docs/source-control/patterns.md (hunk 1 above)
Comment (StarfallProjects) [reply docs/source-control/patterns.md:35]: Added note

File: docs/source-control/git.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+n8n uses this pattern for source control: you'll work with your workflows on your n8n instance, but send them to your Git provider to store them.
+
+## Push, pull, and commit
+
+n8n uses three key Git processes:
+
+* **Push**: send work from your instance to Git. This saves a copy of your workflows, tags, and variables, to Git. You can choose which of these you want to save.
+* **Pull**: get the workflows, tags, and variables from Git and load it into n8n. 
+	!!! warning "Pulling overwrites your work"
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply docs/source-control/git.md:45]: Fixed

File: docs/source-control/using.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Fetch other people's work
+
+To pull work from Git, select **Pull** <span class="inline-image">![Pull icon](/_images/source-control/pull-icon.png)</span> in the main menu.
+
+--8<-- "_snippets/source-control/push-pull-menu-state.md"
+
+n8n may display a warning that:
+
+> Some remote changes are going to override some of your local changes. Are you sure you want to continue?
```
[the remark this answers is not in the document]
Comment (StarfallProjects) [reply docs/source-control/using.md:24]: Updated

**Review by krynble — APPROVED** [review]
Thanks for addressing the stuff I mentioned :)

**Pull request merged**